app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -729,6 +731,75 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml): void
         }
     }
 
+    /**
+     * Replaces strings the PHP date string parser understands with
+     * the actual time in date, datetime and hidden fields.
+     *
+     * Hidden fields must end with `|date` to prevent replacing
+     * strings by a date time representation by accident.
+     */
+    public function populateDateTimeValues(Form $form, string &$formHtml): void
```

Comment:
I'd suggest to add this method to `FormFieldHelper`. It is available in this class and handles various stages of the form including changes based on field types.

Also, I'd ditch the reference:
```suggestion
    private function populateDateTimeValues(Form $form, string $formHtml): string
```
And instead return the modified `$formHtml` as I find it cleaner. It would become a pure function.

I know that you have chosen this because the method `populateField()` does the same thing but it is less confusing if a method has clear input params and a clear output.

Also, can it be `private` if it would be kept in this class?

Reply:
I can move the method and remove the reference (I dislike using `&$var`, anyways).

Can't make it private as it is called from within `FormModel` as the code is. (And I'm a bit hesitant about it, because it makes testing harder :sweat_smile: ).

Reply:
Moved it. This was way too much work for too little to gain IMHO :smile: 

Clean(er) code FTW, though.